dandi/dandiapi.py

```diff
@@ -385,6 +469,24 @@ def _mkasset_from_metadata(self, metadata: Dict[str, Any]) -> "RemoteAsset":
             _metadata=metadata,
         )
 
+    def json_dict(self) -> Dict[str, Any]:
+        """
+        Convert to a JSONable `dict`, omitting the ``client`` attribute and
+        using the same field names as in the API
+        """
+        return {**self._get_data(), "version": self.version.json_dict()}
+
+    def refresh(self) -> None:
+        """Refetch the data for this Dandiset & version"""
+        self._data = self.client.get(f"/dandisets/{self.identifier}/")
+        for vattr in ["most_recent_published_version", "draft_version"]:
+            vdict = self._data.get(vattr)
+            if vdict and vdict["version"] == self.version_id:
+                self._version = Version.parse_obj(vdict)
+                break
+        else:
+            self._version = self.get_version(self.version_id)
```

Comment:
`refresh()` updates the `_data` and also updates the attributes of the `RemoteDandiset`'s `version` attribute, in case anything there has changed.  The version continues to point at the same version as before.  We loop through the version attributes in order to avoid an API call in the event that the `RemoteDandiset`'s version is one of them; if it is neither, we update the version data by refetching it.